app/helpers/repository_helper.rb

```diff
@@ -132,6 +132,14 @@ def add_folder(folder_path, user, repo, path: '/', txn: nil)
 
     folder_path = current_path.join(folder_path)
     folder_path = folder_path.to_s
+
+    # check if only required files are allowed for a submission
+    # allowed folders = paths in required files
+    if required_files.present? && required_files.none? { |file| file.starts_with?(folder_path) }
+      messages << [:extra_files, new_folder_path]
```

Comment:
`new_folder_path` doesn't exist (a bit disappointing that Rubocop didn't pick this up, I might look into that).

Also, please create a new error message for this case rather than reusing the one for `:extra_files`, as the action that triggers this is different.

Reply:
@pretendWhale I think you wrote a test that should fail here? ... can you double check at least one test is hitting this line as well.